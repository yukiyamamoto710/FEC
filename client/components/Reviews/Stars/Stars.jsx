import React from 'react';
import PropTypes from 'prop-types';
import checkStar from '../func/checkStar/checkStar';

const Stars = (props) => {
  const {
    rate,
    name,
    classNameForSize,
    starClick,
  } = props;

  return (
    <div
      data-testid="stars"
      className="ratingHead"
    >
      {[...Array(5)].map((i, index) => (
        <div key={index}>
          <div
            className={`${classNameForSize} ${checkStar(index, rate)}`}
            data-testid={`button${index}`}
            onKeyPress={starClick}
            tabIndex={0}
            role="button"
            label={checkStar(index, rate)}
            name={name}
            id={`${name} ${index + 1}`}
            onClick={starClick}
          />
        </div>
      ))}
    </div>
  );
};

Stars.propTypes = {
  rate: PropTypes.number,
  classNameForSize: PropTypes.string,
  starClick: PropTypes.func,
  name: PropTypes.string,
};

Stars.defaultProps = {
  rate: 1,
  name: '',
  classNameForSize: '',
  starClick: () => 1,
};

export default Stars;
