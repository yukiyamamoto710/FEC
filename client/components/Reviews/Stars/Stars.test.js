/**
 * @jest-environment jsdom
*/
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
  fireEvent,
} from '@testing-library/react';
import Stars from './Stars';

afterEach(cleanup);

it('Stars will trigger when clicked', () => {
  const testFunc = jest.fn();
  const { getByTestId } = render(<Stars starClick={testFunc} />);
  fireEvent.click(getByTestId('button0'));
  fireEvent.click(getByTestId('button1'));
  fireEvent.click(getByTestId('button2'));
  fireEvent.click(getByTestId('button3'));
  fireEvent.click(getByTestId('button4'));
  expect(testFunc).toHaveBeenCalledTimes(5);
});

it('Stars will trigger when keypress', () => {
  const testFunc = jest.fn();
  const { getByTestId } = render(<Stars starClick={testFunc} />);
  fireEvent.keyPress(getByTestId('button0'), { key: 'Enter', code: 13, charCode: 13 });
  fireEvent.keyPress(getByTestId('button1'), { key: 'Enter', code: 13, charCode: 13 });
  fireEvent.keyPress(getByTestId('button2'), { key: 'Enter', code: 13, charCode: 13 });
  fireEvent.keyPress(getByTestId('button3'), { key: 'Enter', code: 13, charCode: 13 });
  fireEvent.keyPress(getByTestId('button4'), { key: 'Enter', code: 13, charCode: 13 });
  expect(testFunc).toHaveBeenCalledTimes(5);
});

it('should have default starClick()', () => {
  expect(Stars.defaultProps.starClick()).toBeDefined();
});

it('matches snapShot', () => {
  const tree = renderer.create(
    <Stars
      rate={1}
      classNameForSize="hi"
      starsClicked={jest.fn}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
