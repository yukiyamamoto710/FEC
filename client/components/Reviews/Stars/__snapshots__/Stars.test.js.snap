// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="ratingHead"
  data-testid="stars"
>
  <div>
    <div
      className="hi fullStar"
      data-testid="button0"
      id=" 1"
      label="fullStar"
      name=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    />
  </div>
  <div>
    <div
      className="hi emptyStar"
      data-testid="button1"
      id=" 2"
      label="emptyStar"
      name=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    />
  </div>
  <div>
    <div
      className="hi emptyStar"
      data-testid="button2"
      id=" 3"
      label="emptyStar"
      name=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    />
  </div>
  <div>
    <div
      className="hi emptyStar"
      data-testid="button3"
      id=" 4"
      label="emptyStar"
      name=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    />
  </div>
  <div>
    <div
      className="hi emptyStar"
      data-testid="button4"
      id=" 5"
      label="emptyStar"
      name=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    />
  </div>
</div>
`;
